### DISCUSSION TITLE
Need "exportAs" in PrimeNG's Angular Directives

### DISCUSSION THREAD

**Discussion by webmatth** [primefaces/community#900] [PrimeNG]
Hi, is it possible to add the "exportAs" parameter in pTooltip and other directives?
like so:
`@Directive({
    selector: '[pTooltip]',
    host: {
        class: 'p-element'
    },
    exportAs: 'pTooltip' <---
})`

read more about this here: https://netbasal.com/angular-2-take-advantage-of-the-exportas-property-81374ce24d26

**Comment by cetincakiroglu** [comment]
Hi,

Why should we add it? I didn't get the point. Users can use tooltips as `[pTooltip]` and `pTooltip`.

**Comment by webmatth** [comment]
Hi Çetin, 
that would allow us to call methods from the pTooltip instance in the template

```
<a pTooltip="I'm a tooltip!!" #tooltip="pTooltip">I'm a link</a>
<button (click)="tooltip.show()">show tooltip</button>
```

without 'exportAs', #tooltip would be seen as an `HTMLAnchorElement`

**Reply by cetincakiroglu** [reply]
Thanks for the explanation, we'll discuss it.

**Reply by dougmbarcellos** [reply]
Hello @cetincakiroglu, any updates?

In my case, I would like to reference the pEditableRow directive, but this is not possible because of the lack of exportAs.
